### PR TITLE
Add ver_datum_code column to bde.crs_work

### PR DESCRIPTION
**Pull request by strk**
See #180

### CODE REVIEW

**Comment by strk** [comment]
I need to know what type the fields need to be. What's the datatype of that field Landonline, @mfwightman / @billgeo ?

**Comment by mfwightman** [comment]
The column exists in the xwrk.crs file, on whichever server Noel Tapp has access to. Before I was getting Noel Tapp to run the queries for me. I should have specified this earlier. Looking at the results he gave me, they are all four letter codes so at a guess CHAR(4).

**Comment by strk** [comment]
Thanks, I see the xwrk.crs file from dataset 20200307214500 having:
`COLUMN  ver_datum_code varchar NULL`
The `SCHEMA` variable in that BDE file is set to `3.21.18`.

The `xwrk.crs` file from dataset 20190720183800 does not have that field, instead, and its `SCHEMA` variable in that BDE file is set to `3.20`.

Can I safely assume that schema is only changed between minor version increments, so that this field was added by Landonline version `3.21` ?

File: sql/99-patches.sql [hunk 1]
```diff
@@ -193,5 +193,50 @@ $$
...
+  END IF;
+
+  -- If table is versioned, use table_version API to add columns
+  IF v_isversioned
+  THEN
+      PERFORM table_version.ver_versioned_table_add_column(
+          'bde',
+          'crs_work',
+          'ver_datum_code',
+          'CHAR(4)'
```
Comment (palmerj) [inline sql/99-patches.sql:229]: VARCHAR(4)? That's how the Landonline database defines it

File: sql/99-patches.sql [hunk 2]
```diff
@@ -193,5 +193,50 @@ $$
...
+      );
+  ELSE
+    -- Otherwise use direct ALTER TABLE
+    ALTER TABLE bde.crs_work ADD COLUMN ver_datum_code CHAR(4);
```
Comment (palmerj) [inline sql/99-patches.sql:233]: As comment above

**Comment by billgeo** [comment]
> Can I safely assume that schema is only changed between minor version increments, so that this field was added by Landonline version 3.21 ?

@iharrison are you able to find out about this from Noel?

**Comment by strk** [comment]
According to Stuart (on slack), `ver_datum_code` was as introduced in 3.20 with change reference `LOLCM-1598 Optimize the code that flags treeview item as SPI dataset`. I don't know why I did not find that column in BDE file from dataset `20190720183800` though, so it could have been done in a patch-level ?

File: sql/99-patches.sql (hunk 1 above)
Comment (strk) [reply sql/99-patches.sql]: I think I used another field as reference which was mapped to CHAR, but ok to change (and changed now) as PostgreSQL size is exactly the same for both types. Note in this schema we've a mix of VARCHAR(x) and CHAR(x) usages.

**Comment by iharrison** [comment]
Yes Noel confirmed it was released in 3.20

**Comment by stuart-datacom** [comment]
> Yes Noel confirmed it was released in 3.20

Have just double checked this - that schema change was actually in 3.21...

_From the db upgrade logs I have traced that schema change script ...
Landonline was upgraded to 3.20.34 on 12 Jul 2019.
That schema change was applied on 22 Nov 2019 in the  3.21.18 release. So was not in 3.20 - was in 3.21.
So yes ....20190720 would be before that column was added.
Our JIRA attributes LOLCM-1598 to 3.20 - but most likely the issue was partially done in 3.20 and was not right or incomplete. So was further amended in 3.21 including that schema change._

**Comment by strk** [comment]
Uff, any subsequent push invalidates reviews, so I'm using my admin powers to merge this after having added CHANGELOG entry

**Pull request merged**